packages/account-management/src/frontend/WellcomeComponents/TextInput.tsx

```diff
@@ -0,0 +1,244 @@
+      }
+    `}
+`;
+
+const TextInputCheckmark = styled.span.attrs({
+  'data-test-id': 'TextInputCheckmark',
+  className: classNames({
+    absolute: true,
+  }),
+})`
+  top: 50%;
+  transform: translateY(-50%);
+  right: 10px;
+  line-height: 0;
+`;
+
+const TextInputErrorMessage = styled.span.attrs({
+  'data-test-id': 'TextInputErrorMessage',
+  className: classNames({
+    'font-hnm': true,
+  }),
+})`
+  font-size: 14px;
+  margin-top: 10px;
+  padding-left: 15px;
+  color: ${props => props.theme.color('red')};
+`;
+
+type Props = {
+  label: string,
+  value: string | undefined,
+  setValue: (value: string) => void,
+  id: string,
+  name?: string,
+  type?: string,
+  pattern?: string,
+  required?: boolean,
+  placeholder?: string,
+  errorMessage?: any,
+  isValid?: boolean,
+  setIsValid?: (value: boolean) => void,
+  showValidity?: boolean,
+  setShowValidity?: (value: boolean) => void,
+  autoFocus?: boolean,
+  big?: boolean,
+  ariaLabel?: string,
+};
+
+// $FlowFixMe (forwardRef)
+const TextInput = forwardRef(
+  (
+    {
+      label,
+      type,
+      value,
+      setValue,
+      id,
+      name,
+      pattern,
+      required,
+      errorMessage,
+      isValid,
+      setIsValid,
+      showValidity,
+      setShowValidity,
+      autoFocus,
+      big,
+      ariaLabel,
+    }: Props,
+    ref // eslint-disable-line
+  ) => {
+    const { isEnhanced } = useContext(AppContext);
+
+    function onChange(event) {
+      const isValueValid = event.currentTarget.validity.valid;
+
+      setValue(event.currentTarget.value);
+
+      if (isValid && setShowValidity) {
+        setShowValidity(false);
+      }
+
+      setIsValid && setIsValid(isValueValid);
+
+      if (isValueValid && setShowValidity) {
+        setShowValidity(false);
+      }
+    }
+
+    function onBlur(event) {
+      setIsValid && setIsValid(event.currentTarget.validity.valid);
+      setShowValidity && setShowValidity(!!value && true);
+    }
+
+    return (
+      <div>
+        <TextInputWrap value={value} hasErrorBorder={!isValid && showValidity} big={big}>
+          <TextInputLabel isEnhanced={isEnhanced} hasValue={!!value} htmlFor={id}>
+            {label}
+          </TextInputLabel>
+          <TextInputInput
+            id={id}
+            name={name}
+            required={required}
+            value={value}
+            pattern={pattern}
+            onChange={onChange}
+            onBlur={onBlur}
+            hasErrorBorder={!isValid && showValidity}
+            type={type}
+            autoFocus={autoFocus}
+            aria-label={ariaLabel}
+          />
+          {isValid && showValidity && (
+            <TextInputCheckmark>
+              <Icon name={`check`} extraClasses={`icon--green`} />
+            </TextInputCheckmark>
+          )}
+        </TextInputWrap>
+        {errorMessage && !isValid && showValidity && (
+          <TextInputErrorMessage dangerouslySetInnerHTML={{ __html: errorMessage }}/>
```

Comment:
Could we adapt this component to accept some JSX as an `errorMessage` instead of an HTML string? Something like:

```tsx
{errorMessage && !isValid && showValidity && (
  <TextInputErrorMessage>
    {errorMessage}
  </TextInputErrorMessage>
)}
```

```ts
// /src/frontend/Registration/Registration.tsx

const emailErrorMessage = () => {
  if (alreadyExists) {
    return (
      <>
        This account already exists. You can try to
        <a href="/">login</a>
      </>
  } else // ...

// ...

<TextInput
  errorMessage={emailErrorMessage()}
/>
```